### DISCUSSION TITLE
How to forward parameters as query string params

### DISCUSSION THREAD

**Discussion by zecar** [adonisjs/core#4441] [Help] [answered]
I have the following situation: I have two controllers, one for Competition, one for CompetitionRound (one competition can have many rounds)

I have the following routes:

GET /competition -> **CompetitionsController.index**
GET /competition/:id -> **CompetitionsController.show**

GET /competition-round -> **CompetitionRoundsController.index**

Now, the index of **CompetitionRoundsController** accepts a query-string parameter for `competitionId` to retrieve only rounds of that competition

I want the same index method to be accessible via
GET /competition/:id/routes

Is there anyway to forward
/competition/`10`/rounds -> /competition-round?`competitionId=10` ?

**Comment by lncitador** [comment]
You can create the route normally in router.method("/competition-round") and in the handler's ctx search for .qs()

See the documentation:
https://docs.adonisjs.com/guides/request#query-string-and-route-params

**Reply by zecar** [reply]
I already have the GET /competition-round where i read request.qs()

My question was how to forward a path parameter as a qs parameter

From /competition/:id/rounds to /competition-round?competitionId=id

**Reply by tomgobich** [reply, marked as answer]
Define route for the path containing the param, then read the params and redirect it using the query string.
https://docs.adonisjs.com/guides/response#redirects

```ts
router.get('/competition/:id/rounds', async ({ response, params }) => {
  return response.redirect().toPath(`/competition-round?competitionId=${params.id}`)
})
```

**Reply by lncitador** [reply]
using named routes.
https://docs.adonisjs.com/guides/response#redirect-with-query-string

**Reply by zecar** [reply]
i am getting an error
![image](https://github.com/adonisjs/core/assets/8117818/4dc33e58-ea7b-4685-8843-22581307f228)

**Reply by brunolipe-a** [reply]
Maybe `router.get`, not `router.on`
